scripts/kani-regression.sh

```diff
@@ -97,6 +97,11 @@ time "$SCRIPT_DIR"/codegen-firecracker.sh
 # Test run 'cargo kani assess scan'
 "$SCRIPT_DIR"/assess-scan-regression.sh
 
+# Test for --manifest-path which we cannot do through compiletest. Also exercise --all-features.
```

Comment:
Question, is this by design or a limitation on how we are organizing our arguments. I.e.: not being able to set these things via manifest file.

I'm asking because we could create new manifest table for these things.

Reply:
It's pretty circular to set the manifest path from the manifest...

There's also no consistent value that could be here. E.g. if we chose the value that would work for compiletest, it'd fail if run with a different working directory.

Reply:
indeed, I was thinking about the --all-features

Reply:
Oh, no that's settable by manifest. In fact, this one isn't doing much here except "exercising" the code: the manifest is setting `--features proof` so it'll work regardless of whether that flag is here or not.

Actually, I didn't introduce `--all-features`, maybe I didn't need to add that here at all. It seems `tests/cargo-kani/feature-flag` exists. I could remove it?

Reply:
Ah, it seems it is required, because our code to pick up kani flags from the manifest file does not correctly find the manifest when using `--manifest-path`.

Reply:
Sorry, I didn't understand this last comment.

Reply:
Removing `--all-features` from that command doesn't work, despite the presence of `feature = ["proof"]` in the manifest file, because the "toml args" hack we use doesn't successfully find the `Cargo.toml` to hack in that argument when we use `--manifest-path`

I'm not sure if we just accept that for now (and open an issue) because the toml args features has always been... less than polished, or if that's a blocker here.

Reply:
I think this will have to be fixed here. There are two problems here:
- We ignore the Kani flags from the given manifest file like you mentioned.
- If I'm not wrong, `cargo kani` could actually pull the Kani flags from the wrong `Cargo.toml`. Since we merge the command arguments and the manifest flags before any parsing.

Reply:
Fair. I was worried that might be a difficult fix, but just parsing arguments twice seems reasonable to do actually... so fixed.